beetmoverscript/script.py

```diff
@@ -86,6 +95,46 @@
     await retry_download(context=context, url=source, path=abs_file_path)
     await retry_upload(context=context, destinations=destinations, path=abs_file_path)
 
+    if update_balrog_manifest:
+        context.balrog_manifest.append(
```

Comment:
Really appreciated the pure non-mutable approach. Good follow-up from yesterday's conversation 👍